src/components/Header.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todos: Todo[];
+  handleTickPressed: () => void;
+  query: string;
+  handleQueryChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  inputRef: React.RefObject<HTMLInputElement>;
+  addTodo: (event: React.FormEvent) => void;
+};
+
+export const Header: React.FC<Props> = ({
+  todos,
+  handleTickPressed,
+  query,
+  handleQueryChange,
+  inputRef,
+  addTodo,
+}) => {
+  return (
+    <header className="todoapp__header">
+      <button
+        type="button"
+        className="todoapp__toggle-all active"
+        data-cy="ToggleAllButton"
+        onClick={handleTickPressed}
```

Comment:
```suggestion
        onClick={handleToggleAllClick}
```
more meaningful name